### DISCUSSION TITLE
Does Popper require the popper element not to have any position styling?

### DISCUSSION THREAD

**Discussion by rohit2sharma95** [floating-ui/floating-ui#1206] [Q&A]
Bootstrap dropdowns use Popper, in a few scenarios they don't use it. After beta1 release and upgrade to Popper V2, users reported some issues, and seems that they are caused by keeping positioning style on the popper element.

**Question:**
If the dropdown element has styles like `right` and `left`, is Popper unable to position it with the reference element correctly then? Or it does not matter what styles you have on the popper element, Popper will always override those styles and place elements correctly with the reference element? Like Popper overrides the margin of the popper element and maintains it by offset.

**Comment by FezVrasta** [comment]
Popper resets `top` and `left`, overriding whatever value may have been set by the user.

**Reply by rohit2sharma95** [reply]
Only `top` and `left` (Not `right` and `bottom`) :thinking: 
And then positions the popper element correctly (If the position of the popper element was wrong by styling)?

**Reply by FezVrasta** [reply]
Also `right` and `bottom`

**Reply by rohit2sharma95** [reply]
Popper may override the values for positioning styles but seems that it can not place the popper element correctly (If the popper element contains any positioning style initially) :thinking: 

These issues are caused because the dropdown has `right: auto` option.
https://github.com/twbs/bootstrap/issues/32484
https://github.com/twbs/bootstrap/issues/32506

**Reply by FezVrasta** [reply]
Ok maybe we don't correctly reset right and bottom then, it's not really something Popper aims to support, the requirement is that the target popper element should not be positioned already.

**Reply by rohit2sharma95** [reply]
Here is the demo (With some modifications in the [official tutorial](https://popper.js.org/docs/v2/tutorial/) from Popper)
https://codepen.io/rohit2sharma95/pen/wvzyGRM

Arrow is placed fine BTW :slightly_smiling_face:

And thanks for your help, can you please also confirm the same requirement was with Popper V1 as well? Did not find it in docs anywhere (Probably should be mentioned in the docs).
